react.md

```diff
@@ -10,37 +10,286 @@ mainmenu: true
+    index.html
+    favicon.ico
+  src/
+    assets/
+      icon.svg
+    containers/
+      App/
+        __tests__/
+          index.test.js
+          SubComponent.test.js
+        index.js
+        SubComponent.js
+        style.module.css
+    components/
+      form/
+      form-partials/
+      layout/
+      ui/
+      view-partials/
+    index.css
+    index.js
+    setupTests.js
+```
+
+### Assets
+Your assets folder should contain any images or SVGs imported by Webpack into your components (CRA's Webpack config only watches `src/`, so you can't import files from `public/`.) Try to put all assets here instead of co-locating them with specific components to reduce possible duplication of assets across multiple components.
+
+#### Importing SVGs
+Webpack allows importing of SVGs as images and as React components. If you need to style an SVG, make use of this feature instead of manually inlining SVG markup into a new component:
+
+```js
+import { ReactComponent as Icon } from 'assets/icon.svg';
+```
+
+### Containers
+Container components differ in that they're directly connected to a data store (whether that's Redux or some other tool) and serve to map data and actions down to child components. An example of a container component would be a `<UserListPage>` component that connects to your store and receives state and dispatch actions as props. For a simple application, a single container may be all you need, while a more complex app may want a container per route. An even more complex application could have multiple containers per page. For example, you might have a complex modal form that's available on multiple pages. It's simpler to make this modal a container and connect it directly to the store than it is to have repetitive prop passing across multiple pages.
+
+For more details on the difference between container & presentational components, refer to [this article by Dan Abramov](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0).
+
+TODO: More here
 
 ### Components
-- TODO: How to organize files within a component, when it's a good time to refactor
-- TODO: How to set up reusable form, layout, and ui components
-- TODO: PropTypes best practices
+While it's tempting to use React as an HTML template engine, it offers more powerful ways for us to develop a set of re-usable & composable components that aren't tied to a specific view or use case. Remember, DRY (**D**on't **R**epeat **Y**ourself) applies to markup and styling as well. This is your opportunity to define a domain-specific markup language for your project, so be descriptive in naming. If you're thorough about building components for all the basic building blocks of your application, you'll find yourself rarely writing raw HTML in your page components. This can feel counter-intuitive at first, but the end result is code that's easier to understand at a glance as you're looking at components that describe what they do instead of large blocks of HTML.
```

Comment:
I think there's a missing closing ** - a lot of this text is bolded after `DRY` ?

Reply:
I think this might be a Github diff bug, the site renders fine & all my bolds are closed.